examples/interchain_queries_test.go

```diff
@@ -0,0 +1,230 @@
+			}},
+		{
+			ChainName: "receiver",
+			ChainConfig: ibc.ChainConfig{
+				Type:           "cosmos",
+				Name:           "receiver",
+				ChainID:        "receiver",
+				Images:         []ibc.DockerImage{dockerImage},
+				Bin:            "icq",
+				Bech32Prefix:   "cosmos",
+				Denom:          "atom",
+				GasPrices:      "0.00atom",
+				TrustingPeriod: "300h",
+				GasAdjustment:  1.1,
+				ModifyGenesis:  modifyGenesisAllowICQQueries([]string{"/cosmos.bank.v1beta1.Query/AllBalances"}), // Add the whitelisted queries to the host chain
+			}},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	chain1, chain2 := chains[0], chains[1]
+
+	// Get a relayer instance
+	r := ibctest.NewBuiltinRelayerFactory(
+		ibc.CosmosRly,
+		zaptest.NewLogger(t),
+		relayer.RelayerOptionExtraStartFlags{Flags: []string{"-p", "events", "-b", "100"}},
+	).Build(t, client, network)
+
+	// Build the network; spin up the chains and configure the relayer
+	const pathName = "test1-test2"
+	const relayerName = "relayer"
+
+	ic := ibctest.NewInterchain().
+		AddChain(chain1).
+		AddChain(chain2).
+		AddRelayer(r, relayerName).
+		AddLink(ibctest.InterchainLink{
+			Chain1:  chain1,
+			Chain2:  chain2,
+			Relayer: r,
+			Path:    pathName,
+		})
+
+	require.NoError(t, ic.Build(ctx, eRep, ibctest.InterchainBuildOptions{
+		TestName:  t.Name(),
+		Client:    client,
+		NetworkID: network,
+
+		SkipPathCreation: false,
+		CreateChannelOpts: ibc.CreateChannelOptions{
+			SourcePortName: "interquery",
+			DestPortName:   "icqhost",
+			Order:          ibc.Unordered,
+			Version:        "icq-1",
+		},
+	}))
+
+	// Fund user accounts, so we can query balances and make assertions.
+	const userFunds = int64(10_000_000_000)
+	users := ibctest.GetAndFundTestUsers(t, ctx, t.Name(), userFunds, chain1, chain2)
+	chain1User := users[0]
+	chain2User := users[1]
+
+	// Wait a few blocks for user accounts to be created on chain.
+	err = test.WaitForBlocks(ctx, 5, chain1, chain2)
+	require.NoError(t, err)
+
+	// Query for the recently created channel-id.
+	channels, err := r.GetChannels(ctx, eRep, chain1.Config().ChainID)
+	require.NoError(t, err)
+
+	// Start the relayer and set the cleanup function.
+	err = r.StartRelayer(ctx, eRep, pathName)
+	require.NoError(t, err)
+
+	t.Cleanup(
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				t.Logf("an error occured while stopping the relayer: %s", err)
+			}
+		},
+	)
+
+	// Wait a few blocks for the relayer to start.
+	err = test.WaitForBlocks(ctx, 5, chain1, chain2)
+	require.NoError(t, err)
+
+	// Query for the balances of an account on the counterparty chain using interchain queries.
+	chanID := channels[0].Counterparty.ChannelID
+	require.NotEqual(t, "", chanID)
+
+	chain1Addr := chain1User.Bech32Address(chain1.Config().Bech32Prefix)
+	require.NotEqual(t, "", chain1Addr)
+
+	chain2Addr := chain2User.Bech32Address(chain2.Config().Bech32Prefix)
+	require.NotEqual(t, "", chain2Addr)
```

Comment:
minor nit:

`require.NotEmpty` or `require.NotZero` may be more appropriate matcher vs `require.NotEqual`.

Reply:
Agreed, gonna change these around real quick.